--- alure-1.2/src/codec_dumb.cpp~	2011-07-29 09:37:48.000000000 +0100
+++ alure-1.2/src/codec_dumb.cpp	2018-12-02 07:40:41.524243606 +0000
@@ -282,7 +282,7 @@
         return -1;
     }
 
-    static long read(char *ptr, long size, void *user_data)
+    static long read(char *ptr, size_t size, void *user_data)
     {
         std::istream *stream = static_cast<dumbStream*>(user_data)->fstream;
         stream->clear();
